File: Tut_03/Insertion_Sort.java
package Tut_03;

public class Insertion_Sort {
	static void insertion_sort(int arr[]) {
		
		for ( int i = 1; i < arr.length; i++) {
			int key = arr[i];
			int j = i - 1;
			
		while ( j >= 0 && arr[j] > key) {
			arr[j + 1] = arr[j];
			j = j - 1;
		}
		arr[j + 1] = key;
			
		}
	}
	public static void main(String[] args) {
		int arr[] = {1,5,2,34,12,4,2,4,5,2};
		insertion_sort(arr);
		for ( int e : arr) {
			System.out.print(e + " ");
		}
	}

}

File: Tut_03/Sorting/Javascript/insertionSort.js
const insertionSort = (a) => {
  let len = a.length;
  let j = 0,
    i = 0;
  let key = 0;
  for (i = 0; i < len; i++) {
    j = i - 1;
    key = a[i];
    while (j >= 0 && a[j] > key) {
      a[j + 1] = a[j];
      j -= 1;
    }
    a[j + 1] = key;
  }
};


const arr = [4,2,25,1,45,21,3,4,132]
insertionSort(arr)
console.log(arr)
